File: 0572-subtree-of-another-tree/0572-subtree-of-another-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSubtree(self, root: Optional[TreeNode], subRoot: Optional[TreeNode]) -> bool:
        def check(a, b):
            if not a and not b:
                return True
            if (a and not b) or (not a and b):
                return False
            if (a.val != b.val):
                return False
            return check(a.left, b.left) and check(a.right, b.right)
        def bfs(node):
            if not node:
                return False
            if (node.val == subRoot.val):
                if check(node, subRoot):
                    return True
            return bfs(node.left) or bfs(node.right)
        return bfs(root)

File: 0572-subtree-of-another-tree/0572-subtree-of-another-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {TreeNode} subRoot
 * @return {boolean}
 */
var isSubtree = function(root, subRoot) {
    const check = (a, b) => {
        if (!a && !b) return true;
        if ((a && !b) || (!a && b)) return false;
        if (a.val !== b.val) return false;
        return check(a.left, b.left) && check(a.right, b.right);
    }
    const bfs = (node) => {
        if (!node) return false;
        if (node.val === subRoot.val && check(node, subRoot)) return true;
        return bfs(node.left) || bfs(node.right);
    }
    return bfs(root);
};